### PR TITLE
Improve developer docs

### PR DESCRIPTION
**Pull request by bajtos**
-  integrate MONOREPO and DEVELOPING guides
- add "new package checklist"
- sort entries in MONOREPO.md
- add content from the wiki page [Contributing](https://github.com/strongloop/loopback-next/wiki/Contributing/96d17f2f328e060a7ca302d43f78be9db83ad4ae)

This is a follow-up for #977 and #1005 and requires https://github.com/strongloop/loopback-next/pull/1013 to be landed first.

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

**Comment by bajtos** [comment]
Preview links:
 - https://github.com/strongloop/loopback-next/blob/docs/improvements/README.md
 - https://github.com/strongloop/loopback-next/blob/docs/improvements/MONOREPO.md
 - https://github.com/strongloop/loopback-next/blob/docs/improvements/CONTRIBUTING.md
 - https://github.com/strongloop/loopback-next/blob/docs/improvements/docs/DEVELOPING.md

File: MONOREPO.md [hunk 1]
```diff
@@ -1,67 +1,28 @@
-# loopback-next
+# Monorepo overview
 
-The [loopback-next](https://github.com/strongloop/loopback-next) repository uses
-[lerna](https://lernajs.io/) to manage multiple packages for LoopBack 4.
+The [loopback-next](https://github.com/strongloop/loopback-next) repository uses [lerna](https://lernajs.io/) to manage multiple packages for LoopBack 4.
 
-## Packages
+<!-- PLEASE KEEP THE TABLE ROWS SORTED ALPHABETICALLY BY PACKAGE NAME-->
```
Comment (raymondfeng) [inline MONOREPO.md]: Hmm, I intentionally sorted the packages by the nature of the modules such as  common utilities, core, extensions, examples.

**Comment by raymondfeng** [comment]
@bajtos Great start. A few comments:

1. The README should have obvious links to other files (developing/monorepo/contributing).
2. Not sure why `DEVELOPING.md` is under `docs` while the rest are not.
3. We probably should have a section in `DEVELOPING.md` to cover how to add documentation for new features.

**Comment by jannyHou** [comment]
@bajtos https://github.com/strongloop/loopback-next/pull/1013 landed 🚢 you can continue yours, thanks!

File: docs/DEVELOPING.md [hunk 2]
```diff
@@ -138,12 +157,54 @@ The **body** provides more details, it should include the motivation for the cha
...
 
+### Tools to help generate a commit message
+
+This repository has [commitizen](https://github.com/commitizen/cz-cli) support enabled. Commitizen can help you generate your commit messages automatically. You must install it globally as follows:
+
+```sh
+$ npm i -g commitizen
+```
+
+And to use it, simply call `git cz` instead of `git commit`. The tool will help you generate a commit message that follows the above guidelines.
```
Comment (richardpringle) [inline docs/DEVELOPING.md]: Were *the above guidelines* directly from `commitizen` or is there some kind of config that sets things like the 100 char "limit".

Might be good to mention where "the above guidelines" come from.

File: docs/DEVELOPING.md [hunk 3]
```diff
@@ -138,12 +157,54 @@ The **body** provides more details, it should include the motivation for the cha
...
+If all steps are successful, it prompts you to publish packages into npm repository.
+
+## New package checklist
+
+When adding (starting) a new package, make sure to complete all steps listed below.
+
+ - Update [MONOREPO.md](../MONOREPO.md) - insert a new table row to describe the new package, please keep the rows sorted by package name.
+ - Update [docs/apidocs.html](./docs/apidocs.html) - add a link to API docs for this new package.
+ - Update [CODEOWNERS](./CODEOWNERS) - add a new entry listing the primary maintainers (owners) of the new package
+ - Ask somebody from the IBM team (e.g. [@bajtos](https://github.com/bajtos), [@raymondfeng](https://github.com/raymondfeng) or [@kjdelisle](https://github.com/kjdelisle)) to enlist the new package on http://apidocs.strongloop.com/
```
Comment (dhmlau) [inline docs/DEVELOPING.md]: http://apidocs.strongloop.com/ -> http://apidocs.loopback.io/

File: docs/DEVELOPING.md [hunk 4]
```diff
@@ -138,12 +157,54 @@ The **body** provides more details, it should include the motivation for the cha
 
 Just as in the subject, use the imperative, present tense: "change" not "changed" nor "changes"a
 
+Paragraphs or bullet points are ok (must not exceed 100 characters per line). Typically a hyphen or asterisk is used for the bullet, followed by a single space, with blank lines in between, but conventions vary here.
```
[the remark this answers is not in the document]
Comment (bajtos) [reply docs/DEVELOPING.md]: Will remove.

File: docs/DEVELOPING.md (hunk 2 above)
Comment (bajtos) [reply docs/DEVELOPING.md]: Most of the guidelines are from https://conventionalcommits.org/, some tweaks are our own (100 char limit, list of scopes defined by monorepo packages). We do mention https://conventionalcommits.org/ at the beginning of _Commit message guidelines_ section. I feel that's good enough.

File: docs/DEVELOPING.md (hunk 3 above)
Comment (bajtos) [reply docs/DEVELOPING.md]: Will fix.

**Comment by bajtos** [comment]
> The README should have obvious links to other files (developing/monorepo/contributing).

Makes sense, will fix.

> Not sure why DEVELOPING.md is under docs while the rest are not.

That's a good question, indeed.

 - I think it's better to keep CONTRIBUTING.md at project root to make it easier for people to find it, as CONTRIBUTING is one of those common files many projects have in project root. Since we are linking to this file from README, I don't mind to move it to `docs`. GitHub's feature showing a link to CONTRIBUTING file supports the `docs` location too, no problem there.

 - I did not move MONOREPO to `docs` because I though it was your intention to have it in project root. If you are ok with the move, then I am happy to move it to `docs`.

@raymondfeng  Thoughts?

> We probably should have a section in DEVELOPING.md to cover how to add documentation for new features.

I agree. Let's leave it out of scope of this pull request though.

>  [Raymond] I intentionally sorted the packages by the nature of the modules such as common utilities, core, extensions, examples.
> [Richard] The problem with "order by nature of modules" is that it's subjective. There's definitely a case to be made for either ordering but I will say that as a "third party", I think it's always easier to reference things that are alphabetically ordered.
> If you are going to group by functionality, you should create separate tables and name the groupings.

@raymondfeng what's your opinion here? May I keep alphabetical order, or are you asking me to preserve your sorting and split the table into multiple ones?

**Comment by bajtos** [comment]
I have rebased the patch on top of the latest master and addressed your review comments.

@raymondfeng @richardpringle LGTY now?

**Pull request merged**